dd-sdk-android-core/src/test/kotlin/com/datadog/android/core/StringBuilderExtKtTest.kt

```diff
@@ -0,0 +1,70 @@
+package com.datadog.android.core
+
+import fr.xgouchet.elmyr.annotation.StringForgery
+import fr.xgouchet.elmyr.junit5.ForgeExtension
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.extension.ExtendWith
+import org.junit.jupiter.api.extension.Extensions
+
+@Extensions(
+    ExtendWith(ForgeExtension::class)
+)
+internal class StringBuilderExtKtTest {
+
+    @Test
+    fun `M add char W addIfNotEmpty {buffer is not empty}`(
+        @StringForgery(regex = ".+") initialContent: String
+    ) {
+        // Given
+        val buffer = StringBuilder(initialContent)
+
+        // When
+        buffer.appendIfNotEmpty(' ')
```

Comment:
nit: ideally you should append a random char, not test it only with `' '` (same for other tests).